pkg/controller/service/service_controller.go

```diff
@@ -582,7 +591,40 @@ func nodeSlicesEqualForLB(x, y []*v1.Node) bool {
 	return nodeNames(x).Equal(nodeNames(x))
 }
 
-func getNodeConditionPredicate() corelisters.NodeConditionPredicate {
+func getNodeTaint() corelisters.NodePredicate {
+	return func(node *v1.Node) bool {
+		// If no taint, accept
+		if len(node.Spec.Taints) == 0 {
```

Comment:
From the discussion with @gmarek and @bgrant0607 we still need to check for `node.spec.unschedulable` see [here](https://github.com/kubernetes/kubernetes/issues/25320#issuecomment-358074388)